backends/cxxrtl/cxxrtl_backend.cc

```diff
@@ -3577,13 +3578,25 @@ struct CxxrtlBackend : public Backend {
 			if (filename == "<stdout>")
 				log_cmd_error("Option -header must be used with a filename.\n");
 
-			worker.intf_filename = filename.substr(0, filename.rfind('.')) + ".h";
-			intf_f.open(worker.intf_filename, std::ofstream::trunc);
+			worker.intf_filepath = filename.substr(0, filename.rfind('.')) + ".h";
+			intf_f.open(worker.intf_filepath, std::ofstream::trunc);
 			if (intf_f.fail())
 				log_cmd_error("Can't open file `%s' for writing: %s\n",
-				              worker.intf_filename.c_str(), strerror(errno));
+				              worker.intf_filepath.c_str(), strerror(errno));
 
 			worker.intf_f = &intf_f;
+
+			// get the base name of the interface file path
+#ifndef _WIN32
+			const std::string sep = "/";
+#else
+			const std::string sep = "\\";
```

Comment:
Both `\\` and `/` are valid separators on Win32.

I suggest making a `basename` function splitting on either and then using it in the `include` directive alone.

Reply:
Thanks for the feedback. I amended the code to take that approach.